src/win/tcp.c

```diff
@@ -48,30 +48,88 @@ static int uv__tcp_nodelay(uv_tcp_t* handle, SOCKET socket, int enable) {
   return 0;
 }
 
+#if defined(TCP_KEEPIDLE) && defined(TCP_KEEPINTVL) && defined(TCP_KEEPCNT)
+/* Windows 10, version 1709 (build 10.0.16299) and later require second units
+ * for TCP keepalive options. */
+#define UV_KEEPALIVE_FACTOR(x)
+#else
+/* On the contrary, Windows <10.0.16299 expect millisecond units. */
+#define UV_KEEPALIVE_FACTOR(x) (x *= 1000)
```

Comment:
Macros should not mutate inputs; `x * 1000` is okay (but you probably want to write `(x)`) but `x *= 1000` is not.

That said, it should be done through run-time detection, not compile-time detection. If I compile on a newer system, it should run without problems on an older system, and vice versa.

Grep src/win/tcp.c for `pRtlGetVersion` for an example.